static/app/components/searchSyntax/renderer.tsx

```diff
@@ -19,30 +19,49 @@ type Props = {
    * highlight active tokens and trigger error tooltips.
    */
   cursorPosition?: number;
+  /**
+   * A function that returns a warning message for a given filter key
+   * will only show a render a warning if the value is truthy
+   */
+  getFilterWarning?: (key: string) => React.ReactNode | undefined;
```

Comment:
I'm not sure if this completely makes sense to be part of the `HighlightQuery`. I'm not a huge fan of this one-off-ish thing being added, especially since Filter tokens are already able to be marked `invalid`.

I am wondering if instead what we should do is add the `getFilterWarning` property to the `SearchConfig` object that is passed when parsing the search query.

https://github.com/getsentry/sentry/blob/6d355406cfe7e38eccb002f6f9aa9df2c66d19ec/static/app/components/searchSyntax/parser.tsx#L868-L871

I could imagine the function basically looks like this

```tsx
  /**
   * A function that returns a warning message for a given filter token
   * will only show a render a warning if the value is truthy
   */
  getFilterTokenWarning?: (keyName: string) => React.ReactNode;
```

(note, `React.ReactNode` already specifies undefined as one of it's types)

Then we add a new property to the resulting filter object type here:

https://github.com/getsentry/sentry/blob/6d355406cfe7e38eccb002f6f9aa9df2c66d19ec/static/app/components/searchSyntax/parser.tsx#L296-L323

Which could be like `warning: this.checkFilterWarning(filter, key, value)`

Which I imagine would just look like

```tsx
/**
 * Checks a filter against some non-grammar validation rules
 */
checkInvalidFilter = <T extends FilterType>(
  filter: T,
  key: FilterMap[T]['key'],
  value: FilterMap[T]['value']
) => {
  const keyName = getKeyName(key);

  return this.config.getFilterTokenWarning(keyName) ?? null;
}

Now `warning` has become a first class citizen on the filter type, so the `showWarning` variable inside of the `FilterToken` simply becomes

```
const showWarning = hasLeft && !!filter.warning;
```

Let me know what you think.